### ISSUE TITLE
gff3validator crashed with error message (see below)

### ISSUE THREAD

**Issue by Rhinogradentia** [genometools/genometools#851]
## Problem description
```
Assertion failed: (!gt_str_cmp(gt_genome_node_get_seqid((GtGenomeNode*) parent), gt_genome_node_get_seqid((GtGenomeNode*) child))), function gt_feature_node_add_child, file src/extended/feature_node.c, line 1099.
This is a bug, please report it at
https://github.com/genometools/genometools/issues
Please make sure you are running the latest release which can be found at
http://genometools.org/pub/
You can check your version number with `gt -version`.
Abgebrochen (Speicherabzug geschrieben)
```
Before the crash it ran without errors and reported gff errors which I've fixed. 

## Exact command line call triggering the problem
`gt gff3validator gff_27032017_small.gff3`

## Example minimal input triggering the problem
A gff3 file to check - repaired some errors of the gff before. Don't know the exact line. 

## What GenomeTools version are you reporting an issue for (as output by `gt -version`)?
```gt -version
gt (GenomeTools) 1.5.10
Copyright (c) 2003-2016 G. Gremme, S. Steinbiss, S. Kurtz, and CONTRIBUTORS
Copyright (c) 2003-2016 Center for Bioinformatics, University of Hamburg
See LICENSE file or http://genometools.org/license.html for license details.

Used compiler: cc (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609
Compile flags:  -g -Wall -Wunused-parameter -pipe -fPIC -Wpointer-arith -Wno-unknown-pragmas -O3 -Werror
```
## Did you compile GenomeTools from source? If so, please state the `make` parameters used.
`make` without parameters

## What operating system (e.g. Ubuntu, Mac OS X), OS version (e.g. 15.10, 10.11) and platform (e.g. x86_64) are you using?

Ubuntu 16.04.4

**Comment by standage** [comment]
Hi Nadine!

Thanks for the bug report. Is there any way you could share the offending file `gff_27032017_small.gff3`? Or a subset thereof that triggers the same error? There's not a whole lot we can do without a bit more info.

**Comment by satta** [comment]
it would also be helpful to see the exact error message, i.e. the message printed before the "This is a bug, please report" line. It seems some assertion failed and it's good to know which one to help pinpoint the problem.

**Comment by satta** [comment]
The assertion seems to suggest that the issue is likely related to a parent-child link between two features with different sequence IDs (GFF3 column 1). However, this case should be reported with a proper error message, and there is code to do that. I'd be curious to find out what GFF3 situation caused this check to be avoided, running into the assertion above. 
Is there any chance that you could share the problematic file?

**Comment by Rhinogradentia** [comment]
Hi, 
I just ran across this error again - I'm sorry that I never came back - I completely forgot about it.

This time it ran on a cluster and not on a Ubuntu machine. The genometools version is:
```
gt -version
gt (GenomeTools) 1.5.10
Copyright (c) 2003-2016 G. Gremme, S. Steinbiss, S. Kurtz, and CONTRIBUTORS
Copyright (c) 2003-2016 Center for Bioinformatics, University of Hamburg
See LICENSE file or http://genometools.org/license.html for license details.

Used compiler: gcc (GCC) 7.3.0
Compile flags: -O2 -ftree-vectorize
[dummy.gff.zip](https://github.com/genometools/genometools/files/8583782/dummy.gff.zip)
 -march=native -fno-math-errno -g -Wall -Wunused-parameter -pipe -fPIC -Wpointer-arith -Wno-unknown-pragmas -O3 -Werror
```
And as nobody else had this error in the meantime I agree that it has something to do with my input file. I don't know if it is still the same file, but it is for sure the same organism. I've minimized it as much as possible and it still produces the error
[dummy.gff.zip](https://github.com/genometools/genometools/files/8583792/dummy.gff.zip)
.
It would just be great if the compromising line would be printed instead of the error. 

Thank you a lot for your help,

Best,
Nadine

**Comment by satta** [comment]
The reason is line 5915 in the file:
```
chr1	XYZ	three_prime_UTR	1906422	1906440	.	+	.	Parent=dummy_Chr3-1069;Name=dummy_Chr3-1069_dummy_Chr3-1069_3-UTR;length=2167;locus_tag=dummy_Chr3-1069;
```
This references a `Parent` (`dummy_Chr3-1069`) that has not yet been seen in the file until later (line 21823) but -- and that's the main issue -- is on a different seqid (`chr3` instead of `chr1`). Usually such a situation would indeed cause a proper error message with the problematic line instead of a failed assertion, but the fact that here it references a yet unseen feature (i.e. is an 'orphan' so far as far as parsing is concerned) triggers a different code path that does not check for this error correctly.

I will look into patching the problem, making the parser more robust for situations like this. Thanks for bringing this to our attention. As for your problem, it looks like the `Parent` attribute here is just wrong. From the context in the file, e.g. the coordinates, I think it should be part of `dummy_Chr1-1057` for example, not something on

**Comment by satta** [comment]
Addressed in merge request #1005. Thanks for the bug report!
